spec/services/control_rate_service_spec.rb

```diff
@@ -246,10 +246,67 @@ def refresh_views
     expect(result[:controlled_patients_rate][Period.month(jan_2020)]).to eq(33)
   end
 
-  it "has a reasonable cache key" do
-    periods = Period.month(july_2018)..Period.month(july_2020)
-    service = ControlRateService.new(facility_group_1, periods: periods)
-    expected_key = "ControlRateService/FacilityGroup/#{facility_group_1.id}/month_periods/#{july_2018.to_date}/#{july_2020.to_date}"
-    expect(service.send(:cache_key)).to eq(expected_key)
+  context "caching" do
+    let(:memory_store) { ActiveSupport::Cache.lookup_store(:redis_store) }
+    let(:cache) { Rails.cache }
```

Comment:
These pieces were the driver for https://github.com/simpledotorg/simple-server/pull/1451